WordPress/Classes/Utility/ImageURLs/WPImageURLHelper.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+
+/// Helper class to create a WordPress URL for various download needs, like specifying image size or using Photon.
+public class WPImageURLHelper: NSObject
+{
+    struct GravatarDefaults {
+        static let scheme = RequestScheme.Secure.rawValue
+        static let host = "secure.\(gravatarURLBase)"
+        // unknownHash = md5("unknown@gravatar.com")
+        static let unknownHash = "ad516503a11cd5ca435acc9bb6523536"
+    }
+
+    enum RequestScheme: String {
+        case Insecure = "http"
+        case Secure = "https"
+    }
+
+    enum URLComponent: String {
+        case Blavatar = "blavatar"
+        case Gravatar = "avatar"
+    }
+
+    enum ImageURLQueryField: String {
```

Comment:
Having second thoughts on whether this should be an `enum` or `struct` with static let's. (Reason: that way you'd avoid the `.rawValue`!). But no big deal, just a matter of taste!